---
title: "Extract major signals from a expression dataset using GenomicSuperSignature"
author: "SehyunOh"
date: "`r Sys.Date()`"
output:
  BiocStyle::html_document:
    toc: true
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval = FALSE
)
```

# Load SRA.avg.loadings
```{r echo=FALSE}
suppressPackageStartupMessages({
  library(GenomicSuperSignature)
})
```

```{r}
library(GenomicSuperSignature)
```

```{r}
load("SRA.avg.loadings")
```

# Load your dataset
```{r}
load("data/test_rseq_dataset")
```

## Pre-processing
Raw gene expression datasets need to be pre-processed.

1. Remove NAs and -Inf/Inf
2. Centered
3. Log2-transformed

## Common genes
```{r}
cm = commonGene()
```

# Extract major signals
## Validate dataset level
```{r}
validationTable = validate()
```

## Assign scores to each sample
```{r}
sampleScore = calScore()
```

# Analyze output
## Main characters of your dataset
```{r}

```

## Sample level characteristics of your dataset
```{r}

```

## Background of the annotation
If there is previous research/paper that backs-up the annotation on your dataset 
and samples, you can find it this way.
```{r}

```











Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
