<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>How to use GenomicSignatures • GenomicSuperSignature</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="How to use GenomicSignatures">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicSuperSignature</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GenomicSignatures.html">How to use GenomicSignatures</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>How to use GenomicSignatures</h1>
                        <h4 class="author">SehyunOh</h4>
            
            <h4 class="date">2020-02-13</h4>
      
      
      <div class="hidden name"><code>GenomicSignatures.Rmd</code></div>

    </div>

    
    
<div id="load-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(SummarizedExperiment)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(GenomicSuperSignature)</a></code></pre></div>
</div>
<div id="list-of-the-available-signature-databases" class="section level1">
<h1 class="hasAnchor">
<a href="#list-of-the-available-signature-databases" class="anchor"></a>List of the available signature databases</h1>
<p>We buit our signature models using recount2 datasets and there are three different versions available with different annotations.</p>
<ol style="list-style-type: decimal">
<li>PLIER pathways (bloodCellMarkersIRISDMAP, canonicalPathways, svmMarkers)</li>
<li>MSigDB_c2c6c7</li>
<li>MSigDB_c6</li>
</ol>
<p>The PLIER pathways are using prior knowledge/ gene sets provided in PLIER package. The MSigDB versions are annotated with different combination of MSigDB gene sets: c2 includes ‘Canonical Pathways’ and ‘Chemical and genetic perturbations’, c6 is ‘Oncogenic signatures’, and c7 is ‘Immunologic signatures’.</p>
<p><br></p>
<p>In this vignette, we are using the signature model annotated with the PLIER pathways, from three gene sets (bloodCellMarkersIRISDMAP, canonicalPathways, svmMarkers).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Size: PLIERmodel ~ 2.1GB, PCAmodel ~ 26.6MB</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">recount_PLIER_model =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(wd, <span class="st">"PLIER/data/recount_PLIER_model.rds"</span>))</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">recount_PCA_model =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(wd, <span class="st">"PCA/data/recount_PCA_model.rds"</span>))</a></code></pre></div>
</div>
<div id="prepare-your-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-your-dataset" class="anchor"></a>Prepare your dataset</h1>
<p>Raw gene expression datasets (= count matrix) need to be pre-processed.</p>
<ol style="list-style-type: decimal">
<li>Convert gene annotation to SYMBOL<br>
</li>
<li>Remove NAs and -Inf/Inf<br>
</li>
<li>Variatn stabilization (log2-transform)</li>
</ol>
<p>We are using TCGA-COAD RNAseq data from <code>curatedTCGAData</code> as an example here. Pre-processing step is described in the vignette, <code>05_Pathway_Preservation</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">"/data2/GenomicSuperSignature/data/TCGA_COAD_rseq.rda"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">TCGA_COAD_rseq =<span class="st"> </span>EnrichmentBrowser<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/idMap.html">idMap</a></span>(TCGA_COAD_rseq, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                                          <span class="dt">org =</span> <span class="st">"hsa"</span>, </a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                                          <span class="dt">from =</span> <span class="st">"ENTREZID"</span>, </a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                                          <span class="dt">to =</span> <span class="st">"SYMBOL"</span>)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">count =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span>(TCGA_COAD_rseq) <span class="op">%&gt;%</span><span class="st"> </span>as.matrix <span class="op">%&gt;%</span><span class="st"> </span>rmNaInf</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">count =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log2</a></span>(count <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)</a></code></pre></div>
</div>
<div id="dataset-level-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#dataset-level-analysis" class="anchor"></a>Dataset level analysis</h1>
<div id="what-are-the-strong-signals-from-my-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#what-are-the-strong-signals-from-my-dataset" class="anchor"></a>What are the strong signals from my dataset?</h2>
<p>Output from <code>validate</code> is the maximum Pearson correlation coefficient between each signature and the top 8 PCs of the test dataset.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">val =<span class="st"> </span><span class="kw"><a href="../reference/validate.html">validate</a></span>(count, recount_PCA_model<span class="op">$</span>avgLoading)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">val[,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</a></code></pre></div>
<pre><code>#&gt; Cl118_01 (38/33) Cl118_02 (22/17)   Cl118_03 (9/5) Cl118_04 (21/15) 
#&gt;       0.27706333       0.19921363       0.09176022       0.03429711</code></pre>
<p><br> You can plot the heatmap of the validation result using <code>heatmapTable</code>. Here, we selected the signatures with &gt; 0.4 correlation coefficient.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="../reference/heatmapTable.html">heatmapTable</a></span>(val[, <span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(val <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.4</span>), <span class="dt">drop =</span> <span class="ot">FALSE</span>], </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">             <span class="dt">row_title =</span> <span class="st">"COAD"</span>, </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">             <span class="dt">column_title =</span> <span class="st">"Validation using recount_PCAmodel"</span>,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">             <span class="dt">colors =</span> col)</a></code></pre></div>
<p><img src="GenomicSignatures_files/figure-html/validate_heatmap-1.png" width="700"></p>
</div>
<div id="which-studies-are-similar-to-your-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#which-studies-are-similar-to-your-dataset" class="anchor"></a>Which studies are similar to your dataset?</h2>
<p>For example, if you want to check which studies contributed cluster 53, you can use <code>findStudiesInCluster</code> function.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="../reference/findStudiesInCluster.html">findStudiesInCluster</a></span>(recount_PCA_model, <span class="dv">53</span>)</a></code></pre></div>
<pre><code>#&gt;  [1] "SRP001540" "SRP002079" "SRP003611" "SRP004637" "SRP007461" "SRP014320"
#&gt;  [7] "SRP014428" "SRP019994" "SRP024274" "SRP026042" "SRP026126" "SRP026537"
#&gt; [13] "SRP028155" "SRP028594" "SRP029262" "SRP032775" "SRP033725" "SRP039077"
#&gt; [19] "SRP039397" "SRP041036" "SRP042286" "SRP042596" "SRP042620" "SRP047233"
#&gt; [25] "SRP047476" "SRP049340" "SRP050272" "SRP050892" "SRP052740" "SRP056197"
#&gt; [31] "SRP056477" "SRP056696" "SRP057205" "SRP057758" "SRP058856" "SRP061881"
#&gt; [37] "SRP062025" "SRP063493" "SRP063496" "SRP064259" "SRP060355" "SRP036848"
#&gt; [43] "SRP041538"</code></pre>
</div>
<div id="which-gene-sets-pathways-are-related-to-the-signal" class="section level2">
<h2 class="hasAnchor">
<a href="#which-gene-sets-pathways-are-related-to-the-signal" class="anchor"></a>Which gene sets/ pathways are related to the signal?</h2>
<p>Not all signatures are annotated with gene sets/pathways - it varies depending on signature database you selected. You can check the list of annotated signatures.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(recount_PCA_model<span class="op">$</span>annotation)</a></code></pre></div>
<pre><code>#&gt;  [1] "Cl118_02 (22/17)"  "Cl118_05 (33/30)"  "Cl118_09 (45/33)" 
#&gt;  [4] "Cl118_11 (10/9)"   "Cl118_13 (14/10)"  "Cl118_14 (34/27)" 
#&gt;  [7] "Cl118_19 (33/30)"  "Cl118_20 (20/15)"  "Cl118_24 (36/27)" 
#&gt; [10] "Cl118_27 (75/50)"  "Cl118_28 (8/5)"    "Cl118_31 (49/40)" 
#&gt; [13] "Cl118_37 (20/17)"  "Cl118_38 (30/26)"  "Cl118_39 (25/20)" 
#&gt; [16] "Cl118_44 (16/15)"  "Cl118_45 (62/46)"  "Cl118_49 (13/11)" 
#&gt; [19] "Cl118_55 (58/48)"  "Cl118_59 (21/15)"  "Cl118_63 (21/20)" 
#&gt; [22] "Cl118_64 (71/47)"  "Cl118_65 (10/8)"   "Cl118_69 (83/52)" 
#&gt; [25] "Cl118_72 (32/24)"  "Cl118_74 (190/78)" "Cl118_75 (11/8)"  
#&gt; [28] "Cl118_76 (23/17)"  "Cl118_82 (21/19)"  "Cl118_84 (35/29)" 
#&gt; [31] "Cl118_92 (15/12)"  "Cl118_93 (8/8)"    "Cl118_94 (9/7)"   
#&gt; [34] "Cl118_95 (10/9)"   "Cl118_98 (50/43)"  "Cl118_100 (37/33)"
#&gt; [37] "Cl118_102 (35/34)" "Cl118_106 (12/6)"  "Cl118_107 (52/43)"
#&gt; [40] "Cl118_108 (33/29)" "Cl118_109 (46/35)" "Cl118_110 (27/21)"
#&gt; [43] "Cl118_111 (36/30)" "Cl118_112 (16/14)" "Cl118_113 (19/17)"
#&gt; [46] "Cl118_114 (66/39)" "Cl118_118 (19/15)"</code></pre>
<p><br> The gene sets used in the GSEA is,</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">recount_PCA_model<span class="op">$</span>geneSets</a></code></pre></div>
<pre><code>#&gt; [1] "Three priors from PLIER package: bloodCellMarkersIRISDMAP, canonicalPathways, svmMarkers"</code></pre>
<p><br> You can subset the top and bottum <code>n</code> pathways enriched in GSEA.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">n =<span class="st"> </span><span class="dv">10</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">pathwaySubsets =<span class="st"> </span><span class="kw">subsetAnnotatedPathways</span>(recount_PCA_model<span class="op">$</span>annotation, n)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">pathwaySubsets[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<pre><code>#&gt;                                                 Cl118_02 (22/17)
#&gt; Up_1                                            MIPS_SPLICEOSOME
#&gt; Up_2                                    REACTOME_MRNA_PROCESSING
#&gt; Up_3    REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA
#&gt; Up_4                              REACTOME_MITOTIC_M_M_G1_PHASES
#&gt; Up_5                                      REACTOME_MRNA_SPLICING
#&gt; Up_6                                         REACTOME_CELL_CYCLE
#&gt; Up_7                                   REACTOME_G2_M_CHECKPOINTS
#&gt; Up_8                                    REACTOME_DNA_REPLICATION
#&gt; Up_9                             REACTOME_CHROMOSOME_MAINTENANCE
#&gt; Up_10                                        PID_FANCONI_PATHWAY
#&gt; Down_1                                        IRIS_Monocyte-Day1
#&gt; Down_2                                                DMAP_GRAN2
#&gt; Down_3                                                DMAP_MONO2
#&gt; Down_4                           KEGG_HEMATOPOIETIC_CELL_LINEAGE
#&gt; Down_5                            PID_A6B1_A6B4_INTEGRIN_PATHWAY
#&gt; Down_6                                               DMAP_DENDA2
#&gt; Down_7                                                DMAP_MONO1
#&gt; Down_8                       REACTOME_INTERFERON_GAMMA_SIGNALING
#&gt; Down_9                          REACTOME_CELL_CELL_COMMUNICATION
#&gt; Down_10                                       PID_HIF1_TFPATHWAY
#&gt;                                                 Cl118_05 (33/30)
#&gt; Up_1                                    REACTOME_DNA_REPLICATION
#&gt; Up_2                              REACTOME_MITOTIC_M_M_G1_PHASES
#&gt; Up_3                               REACTOME_MITOTIC_PROMETAPHASE
#&gt; Up_4                                 REACTOME_CELL_CYCLE_MITOTIC
#&gt; Up_5                                         REACTOME_CELL_CYCLE
#&gt; Up_6                             REACTOME_CELL_CYCLE_CHECKPOINTS
#&gt; Up_7                                   REACTOME_SYNTHESIS_OF_DNA
#&gt; Up_8                             REACTOME_MITOTIC_G1_G1_S_PHASES
#&gt; Up_9                                    REACTOME_M_G1_TRANSITION
#&gt; Up_10   REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA
#&gt; Down_1                         KEGG_CELL_ADHESION_MOLECULES_CAMS
#&gt; Down_2                                                 DMAP_NKA1
#&gt; Down_3                                        SVM Macrophages M0
#&gt; Down_4                                        SVM Macrophages M1
#&gt; Down_5                                    SVM Mast cells resting
#&gt; Down_6                                              DMAP_TCELLA2
#&gt; Down_7                                               DMAP_DENDA2
#&gt; Down_8                                                DMAP_GRAN2
#&gt; Down_9                                                DMAP_MONO1
#&gt; Down_10                         KEGG_ARACHIDONIC_ACID_METABOLISM</code></pre>
</div>
</div>
<div id="sample-level-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-level-analysis" class="anchor"></a>Sample level analysis</h1>
<div id="what-are-the-strong-signals-from-my-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#what-are-the-strong-signals-from-my-sample" class="anchor"></a>What are the strong signals from my sample?</h2>
<p>You can assigne scores to an individual samples using <code>calScore</code>. In the following example, each sample (total 524 in this example) is assigned with the scores from all signatures (118 in this example).</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">score =<span class="st"> </span><span class="kw"><a href="../reference/calScore.html">calScore</a></span>(count, recount_PCA_model<span class="op">$</span>avgLoading)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(score[[<span class="dv">1</span>]])</a></code></pre></div>
<pre><code>#&gt; [1] 524 118</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">score[[<span class="dv">1</span>]][<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<pre><code>#&gt;                              Cl118_01 (38/33) Cl118_02 (22/17)
#&gt; TCGA-DM-A28G-01A-11R-A16W-07        -6.959946         7.939491
#&gt; TCGA-AA-A004-01A-01R-A00A-07        20.071055        -5.501803
#&gt; TCGA-CK-4950-01A-01R-1723-07       -13.105493         3.748127
#&gt; TCGA-D5-5539-01A-01R-1653-07        -5.265413         2.476548</code></pre>
<p><br></p>
<p>Score result can be displayed in a heatmap format using <code>sampleScoreHeatmap</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="../reference/sampleScoreHeatmap.html">sampleScoreHeatmap</a></span>(<span class="dt">score =</span> score[[<span class="dv">1</span>]], </a>
<a class="sourceLine" id="cb19-2" data-line-number="2">                   <span class="dt">dataName =</span> <span class="st">"TCGA_COAD_RNAseq"</span>, </a>
<a class="sourceLine" id="cb19-3" data-line-number="3">                   <span class="dt">modelName =</span> <span class="st">"recount_PCAmodel"</span>)</a></code></pre></div>
<p><img src="GenomicSignatures_files/figure-html/sampleScoreHeatmap-1.png" width="700"></p>
<p><br></p>
<p>You can plot a specific sample.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="../reference/sampleScoreHeatmap.html">sampleScoreHeatmap</a></span>(<span class="dt">score =</span> score[[<span class="dv">1</span>]][<span class="dv">52</span><span class="op">:</span><span class="dv">54</span>,], </a>
<a class="sourceLine" id="cb20-2" data-line-number="2">                   <span class="dt">dataName =</span> <span class="st">"TCGA_COAD_RNAseq"</span>, </a>
<a class="sourceLine" id="cb20-3" data-line-number="3">                   <span class="dt">modelName =</span> <span class="st">"recount_PCAmodel"</span>,</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">                   <span class="dt">row_names_gp =</span> <span class="dv">7</span>)</a></code></pre></div>
<p><img src="GenomicSignatures_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="what-are-the-similar-studies-and-related-pathways-to-the-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#what-are-the-similar-studies-and-related-pathways-to-the-sample" class="anchor"></a>What are the similar studies and related pathways to the sample?</h2>
<p>You can find the simlar studies and the related pathways in a same way as in ‘Dataset level analysis’ above (Section 4.2 and 4.3).</p>
<p><br></p>
<div id="example" class="section level3">
<h3 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h3>
<p>There are three tumor (<code>01</code>) samples from ‘TCGA-A6-6650’ patient prepared in two different vials (<code>A</code> and <code>B</code>).</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">sampleID =<span class="st"> "TCGA-A6-6650"</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2">ind =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(sampleID, <span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(TCGA_COAD_rseq)<span class="op">$</span>sample)</a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(TCGA_COAD_rseq)[ind,]</a></code></pre></div>
<pre><code>#&gt; DataFrame with 3 rows and 3 columns
#&gt;                                                    sample     type     GROUP
#&gt;                                               &lt;character&gt; &lt;factor&gt; &lt;numeric&gt;
#&gt; TCGA-A6-6650-01B-02R-A277-07 TCGA-A6-6650-01B-02R-A277-07     COAD         1
#&gt; TCGA-A6-6650-01A-11R-1774-07 TCGA-A6-6650-01A-11R-1774-07     COAD         1
#&gt; TCGA-A6-6650-01A-11R-A278-07 TCGA-A6-6650-01A-11R-A278-07     COAD         1</code></pre>
<p><br></p>
<p>It seems like the different vial (A and B) results in a batch effect.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw"><a href="../reference/sampleScoreHeatmap.html">sampleScoreHeatmap</a></span>(<span class="dt">score =</span> score[[<span class="dv">1</span>]][ind,], </a>
<a class="sourceLine" id="cb23-2" data-line-number="2">                   <span class="dt">dataName =</span> <span class="st">"TCGA_COAD_RNAseq"</span>, </a>
<a class="sourceLine" id="cb23-3" data-line-number="3">                   <span class="dt">modelName =</span> <span class="st">"recount_PCAmodel"</span>,</a>
<a class="sourceLine" id="cb23-4" data-line-number="4">                   <span class="dt">row_names_gp =</span> <span class="dv">7</span>)</a></code></pre></div>
<p><img src="GenomicSignatures_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p><br></p>
<p>Check the similar studies and related pathways of ‘TCGA-A6-6650-01B-02R-A277-07’.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">ind =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(TCGA_COAD_rseq)<span class="op">$</span>sample <span class="op">==</span><span class="st"> "TCGA-A6-6650-01B-02R-A277-07"</span>)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(score[[<span class="dv">1</span>]][ind,])</a></code></pre></div>
<pre><code>#&gt; Cl118_53 (47/43) 
#&gt;               53</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/which.min.html">which.min</a></span>(score[[<span class="dv">1</span>]][ind,])</a></code></pre></div>
<pre><code>#&gt; Cl118_58 (26/26) 
#&gt;               58</code></pre>
<p><br></p>
<p>Signature 53 (<code>Cl118_53</code>) doesn’t have enriched gene sets, so we are checking only the similar studies.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="../reference/findStudiesInCluster.html">findStudiesInCluster</a></span>(recount_PCA_model, <span class="dv">53</span>)</a></code></pre></div>
<pre><code>#&gt;  [1] "SRP001540" "SRP002079" "SRP003611" "SRP004637" "SRP007461" "SRP014320"
#&gt;  [7] "SRP014428" "SRP019994" "SRP024274" "SRP026042" "SRP026126" "SRP026537"
#&gt; [13] "SRP028155" "SRP028594" "SRP029262" "SRP032775" "SRP033725" "SRP039077"
#&gt; [19] "SRP039397" "SRP041036" "SRP042286" "SRP042596" "SRP042620" "SRP047233"
#&gt; [25] "SRP047476" "SRP049340" "SRP050272" "SRP050892" "SRP052740" "SRP056197"
#&gt; [31] "SRP056477" "SRP056696" "SRP057205" "SRP057758" "SRP058856" "SRP061881"
#&gt; [37] "SRP062025" "SRP063493" "SRP063496" "SRP064259" "SRP060355" "SRP036848"
#&gt; [43] "SRP041538"</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw"><a href="../reference/findStudiesInCluster.html">findStudiesInCluster</a></span>(recount_PCA_model, <span class="dv">38</span>)</a></code></pre></div>
<pre><code>#&gt;  [1] "SRP002079" "SRP007461" "SRP007825" "SRP011546" "SRP026126" "SRP028291"
#&gt;  [7] "SRP028336" "SRP029880" "SRP033248" "SRP035988" "SRP039397" "SRP041471"
#&gt; [13] "SRP042286" "SRP042620" "SRP044668" "SRP048801" "SRP049593" "SRP049988"
#&gt; [19] "SRP050892" "SRP051083" "SRP051844" "SRP053246" "SRP055810" "SRP058181"
#&gt; [25] "SRP063496" "SRP060355"</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">pathwaySubsets[<span class="st">"Cl118_09 (45/33)"</span>]</a></code></pre></div>
<pre><code>#&gt;                                                                Cl118_09 (45/33)
#&gt; Up_1                                                                  DMAP_ERY3
#&gt; Up_2                                                                  DMAP_ERY5
#&gt; Up_3                               REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM
#&gt; Up_4                                                                  DMAP_ERY4
#&gt; Up_5                                                     REACTOME_TRANSCRIPTION
#&gt; Up_6                                                 REACTOME_METABOLISM_OF_RNA
#&gt; Up_7                                                          PID_PDGFRBPATHWAY
#&gt; Up_8                                            REACTOME_MITOTIC_G1_G1_S_PHASES
#&gt; Up_9                                              REACTOME_MEMBRANE_TRAFFICKING
#&gt; Up_10                                       KEGG_NEUROTROPHIN_SIGNALING_PATHWAY
#&gt; Down_1  REACTOME_TRANSPORT_OF_MATURE_MRNA_DERIVED_FROM_AN_INTRONLESS_TRANSCRIPT
#&gt; Down_2                     REACTOME_TRANSPORT_OF_MATURE_TRANSCRIPT_TO_CYTOPLASM
#&gt; Down_3                                               REACTOME_GLUCOSE_TRANSPORT
#&gt; Down_4                 REACTOME_INTERACTIONS_OF_VPR_WITH_HOST_CELLULAR_PROTEINS
#&gt; Down_5                                                          PID_ATR_PATHWAY
#&gt; Down_6                                                        PID_NFAT_3PATHWAY
#&gt; Down_7                                            REACTOME_MITOTIC_PROMETAPHASE
#&gt; Down_8                                                     PID_PI3KCIAKTPATHWAY
#&gt; Down_9                                                        PID_GMCSF_PATHWAY
#&gt; Down_10                                                 REACTOME_IL_2_SIGNALING</code></pre>
<p><br></p>
<p>Check the similar studies and related pathways of ‘TCGA-A6-6650-01A-11R-1774-07’.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">ind =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span>(TCGA_COAD_rseq)<span class="op">$</span>sample <span class="op">==</span><span class="st"> "TCGA-A6-6650-01A-11R-1774-07"</span>)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(score[[<span class="dv">1</span>]][ind,])</a></code></pre></div>
<pre><code>#&gt; Cl118_38 (30/26) 
#&gt;               38</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/which.min.html">which.min</a></span>(score[[<span class="dv">1</span>]][ind,])</a></code></pre></div>
<pre><code>#&gt; Cl118_09 (45/33) 
#&gt;                9</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="kw"><a href="../reference/findStudiesInCluster.html">findStudiesInCluster</a></span>(recount_PCA_model, <span class="dv">38</span>)</a></code></pre></div>
<pre><code>#&gt;  [1] "SRP002079" "SRP007461" "SRP007825" "SRP011546" "SRP026126" "SRP028291"
#&gt;  [7] "SRP028336" "SRP029880" "SRP033248" "SRP035988" "SRP039397" "SRP041471"
#&gt; [13] "SRP042286" "SRP042620" "SRP044668" "SRP048801" "SRP049593" "SRP049988"
#&gt; [19] "SRP050892" "SRP051083" "SRP051844" "SRP053246" "SRP055810" "SRP058181"
#&gt; [25] "SRP063496" "SRP060355"</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">pathwaySubsets[<span class="st">"Cl118_38 (30/26)"</span>]</a></code></pre></div>
<pre><code>#&gt;                                               Cl118_38 (30/26)
#&gt; Up_1                                      PID_PI3KCIAKTPATHWAY
#&gt; Up_2    REACTOME_PYRUVATE_METABOLISM_AND_CITRIC_ACID_TCA_CYCLE
#&gt; Up_3                        REACTOME_SIGNAL_TRANSDUCTION_BY_L1
#&gt; Up_4     REACTOME_AMINO_ACID_AND_OLIGOPEPTIDE_SLC_TRANSPORTERS
#&gt; Up_5                              REACTOME_TRNA_AMINOACYLATION
#&gt; Up_6                                KEGG_N_GLYCAN_BIOSYNTHESIS
#&gt; Up_7                REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION
#&gt; Up_8                                       PID_AJDISS_2PATHWAY
#&gt; Up_9             REACTOME_MAP_KINASE_ACTIVATION_IN_TLR_CASCADE
#&gt; Up_10                         KEGG_AMINOACYL_TRNA_BIOSYNTHESIS
#&gt; Down_1          REACTOME_METABOLISM_OF_LIPIDS_AND_LIPOPROTEINS
#&gt; Down_2                                               DMAP_ERY3
#&gt; Down_3                                               DMAP_ERY4
#&gt; Down_4                                               DMAP_ERY5
#&gt; Down_5                        REACTOME_PHOSPHOLIPID_METABOLISM
#&gt; Down_6      REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES
#&gt; Down_7                              REACTOME_SIGNALLING_BY_NGF
#&gt; Down_8     REACTOME_TRANSMEMBRANE_TRANSPORT_OF_SMALL_MOLECULES
#&gt; Down_9                    REACTOME_METABOLISM_OF_CARBOHYDRATES
#&gt; Down_10                  REACTOME_SIGNALING_BY_FGFR_IN_DISEASE</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw"><a href="../reference/findStudiesInCluster.html">findStudiesInCluster</a></span>(recount_PCA_model, <span class="dv">9</span>)</a></code></pre></div>
<pre><code>#&gt;  [1] "SRP002079" "SRP007525" "SRP009408" "SRP015640" "SRP020492" "SRP021193"
#&gt;  [7] "SRP025982" "SRP026126" "SRP027383" "SRP028336" "SRP030617" "SRP033566"
#&gt; [13] "SRP042286" "SRP044668" "SRP047476" "SRP049593" "SRP049820" "SRP050223"
#&gt; [19] "SRP050272" "SRP050992" "SRP051083" "SRP051472" "SRP053246" "SRP057065"
#&gt; [25] "SRP057500" "SRP057758" "SRP058181" "SRP059057" "SRP059775" "SRP061881"
#&gt; [31] "SRP063840" "SRP065812" "SRP036848"</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">pathwaySubsets[<span class="st">"Cl118_09 (45/33)"</span>]</a></code></pre></div>
<pre><code>#&gt;                                                                Cl118_09 (45/33)
#&gt; Up_1                                                                  DMAP_ERY3
#&gt; Up_2                                                                  DMAP_ERY5
#&gt; Up_3                               REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM
#&gt; Up_4                                                                  DMAP_ERY4
#&gt; Up_5                                                     REACTOME_TRANSCRIPTION
#&gt; Up_6                                                 REACTOME_METABOLISM_OF_RNA
#&gt; Up_7                                                          PID_PDGFRBPATHWAY
#&gt; Up_8                                            REACTOME_MITOTIC_G1_G1_S_PHASES
#&gt; Up_9                                              REACTOME_MEMBRANE_TRAFFICKING
#&gt; Up_10                                       KEGG_NEUROTROPHIN_SIGNALING_PATHWAY
#&gt; Down_1  REACTOME_TRANSPORT_OF_MATURE_MRNA_DERIVED_FROM_AN_INTRONLESS_TRANSCRIPT
#&gt; Down_2                     REACTOME_TRANSPORT_OF_MATURE_TRANSCRIPT_TO_CYTOPLASM
#&gt; Down_3                                               REACTOME_GLUCOSE_TRANSPORT
#&gt; Down_4                 REACTOME_INTERACTIONS_OF_VPR_WITH_HOST_CELLULAR_PROTEINS
#&gt; Down_5                                                          PID_ATR_PATHWAY
#&gt; Down_6                                                        PID_NFAT_3PATHWAY
#&gt; Down_7                                            REACTOME_MITOTIC_PROMETAPHASE
#&gt; Down_8                                                     PID_PI3KCIAKTPATHWAY
#&gt; Down_9                                                        PID_GMCSF_PATHWAY
#&gt; Down_10                                                 REACTOME_IL_2_SIGNALING</code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#load-packages">Load packages</a></li>
      <li><a href="#list-of-the-available-signature-databases">List of the available signature databases</a></li>
      <li><a href="#prepare-your-dataset">Prepare your dataset</a></li>
      <li>
<a href="#dataset-level-analysis">Dataset level analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#what-are-the-strong-signals-from-my-dataset">What are the strong signals from my dataset?</a></li>
      <li><a href="#which-studies-are-similar-to-your-dataset">Which studies are similar to your dataset?</a></li>
      <li><a href="#which-gene-sets-pathways-are-related-to-the-signal">Which gene sets/ pathways are related to the signal?</a></li>
      </ul>
</li>
      <li>
<a href="#sample-level-analysis">Sample level analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#what-are-the-strong-signals-from-my-sample">What are the strong signals from my sample?</a></li>
      <li><a href="#what-are-the-similar-studies-and-related-pathways-to-the-sample">What are the similar studies and related pathways to the sample?</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sehyun Oh.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
